### PR TITLE
Reload unicorn if configs have been updated during provision

### LINKED ISSUE #380: Reload unicorn when settings have changed
As noted here: https://github.com/openfoodfoundation/ofn-install/pull/356#issuecomment-480106220 we don't reload unicorn configurations when the settings have been changed during provisioning.

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Closes #380 

Reloads unicorn if it's configuration has been changed during provision.

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
@sauloperez Yes, this PR prompted me to look at doing a refactor of handlers and making a reusable `shared_handlers` section to include in multiple playbooks, I'm taking at it look now...

**Comment by Matt-Yorkley** [comment]
I've updated this to use a graceful reload instead of a harsh restart.

**Comment by sauloperez** [comment]
But does unicorn's reload read the config again?

**Comment by Matt-Yorkley** [comment]
The unicorn process itself supports a `reload` option, which "reloads config file and gracefully restart all workers" (https://bogomips.org/unicorn.git/tree/SIGNALS) if the `HUP` signal is passed to the main process.

Our custom systemd unit that we use for controlling the process was not set up to properly support the `reload` option.

**Comment by sauloperez** [comment]
That's great news! Reload is the biggest upside of Unicorn and it is a shame we were not using it (although we knew).

File: roles/webserver/handlers/main.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- name: reload unicorn
+  service_facts:
+  notify: reload unicorn if running
+
+- name: reload unicorn if running
+  service:
+    name: unicorn_{{ app }}
+    state: reloaded
+  become: yes
+  when: ansible_facts.services['unicorn_openfoodnetwork.service']['state'] == 'running'
```
Comment (sauloperez) [inline roles/webserver/handlers/main.yml]: :ok_hand:

**Pull request merged**